feat:  회원가입 UI 구현

soo96 opened the pull request:
## 요약

- 회원가입 UI 구현

## 변경 사항
- [x]  회원가입 페이지 생성
    - [x]  닉네임 입력 인풋 생성
    - [x] [가입하기] 버튼 생성
      - [x]  [가입하기] 버튼 클릭 시, 닉네임 중복여부 응답.
    - [x] 닉네임 중복 시 "중복된 닉네임입니다." 에러메시지 표시되어야 한다.
    - [x] 정상 가입 시 회원가입 완료 페이지로 이동.
- [x]  회원가입 완료 페이지 생성
    - [x]  [처음으로] 버튼 생성
        - [x]  [처음으로] 버튼 클릭 시, 로그인 화면으로 이동하게 된다.


## 체크리스트

- [ ] 이해하기 어려운 코드에는 주석을 추가했습니다  
- [ ] 관련 문서를 수정했습니다  
- [ ] 새로운 경고가 발생하지 않습니다  
- [ ] 수정한 기능 또는 추가한 기능에 대해 테스트를 작성했습니다  
- [ ] 의존성 있는 변경 사항이 병합 및 배포되었습니다

## 관련 이슈

<!-- 이 PR이 해결하는 관련 이슈나 버그가 있다면 연결해 주세요. 어떤 문제를 해결하는 코드인지 맥락을 제공합니다. -->

## 리뷰 반영 사항

<!-- PR리뷰 반형 사항 -->

<details>
<summary><strong>선택 항목 펼치기</strong></summary>

## 스크린샷

<!-- 시각적인 변경 사항이 있다면 스크린샷이나 GIF를 포함해 주세요. 리뷰어가 쉽게 이해할 수 있습니다. -->

## 테스트 방법

<!-- PR에서 수행한 변경 사항을 어떻게 테스트할 수 있는지 설명해 주세요. 리뷰어가 직접 확인할 수 있도록 돕습니다. -->

## 리뷰어 참고사항

<!-- 리뷰어에게 특별히 알려야 할 사항이나 고려할 점이 있다면 여기에 작성해 주세요. -->

</details>

soo96 committed:
chore: dev브랜치 최신 사항 반영

refactor: 로그인 페이지 UI 개선

soo96 committed:
feat: Signup, SignupComplete 페이지 정적구현

soo96 committed:
refactor: ProgressBar에 half/full 구분을 위한 progress prop 추가

soo96 committed:
chore: 에러 메시지 영역 높이 고정

soo96 committed:
refactor: 회원가입 처리 로직 useSignupMutation 훅으로 분리

soo96 committed:
feat: Signup, SignupComplete 컴포넌트 구현 완료

soo96 committed:
chore: dev브랜치 최신 사항 반영

smileboy0014 commented on src/components/signup/ProgressBar.tsx:
```diff
@@ -0,0 +1,21 @@
+
+interface ProgressBarType {
+  progress: ProgressType;
+}
```
저희 props 정의는 type으로 하는게 어떨까요?
- https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example/#more-advice
여기를 보시면, props를 정의할 때는 type을 더 고려해봐라 라는 말이 있는 것 같습니다 😀

smileboy0014 commented on src/pages/Signup.tsx:
```diff
@@ -0,0 +1,69 @@
+  const { requestData, user } = location.state;
+
+  const onErrorCallback = (errorCode: string) => {
+    if (errorCode === '3401') {
+      setErrorMessage('중복된 닉네임입니다.');
```
저희 에러 코드 정해둔걸로 알고 있는데, 따로 에러코드에 대한 메세지를 정의해둬야 하지 않을까요?

soo96 replied:
프론트도 추후에 일괄적으로 에러코드, 에러메시지 정의해서 반영하면 될 것 같습니다.🙂

smileboy0014 commented on src/pages/Signup.tsx:
```diff
@@ -0,0 +1,69 @@
+      <SignupButton
+        text={isPending ? '가입중...' : '가입하기'}
+        onClick={() => {
+          handleSignup();
```
```suggestion
        onClick={() => {
          handleClick();
```
제가 좀 조사해본바로는 on[EventName]과 매칭되는게 handle[EvnetName]식으로, 처리하는걸로 알아봤습니다.
- https://stackoverflow.com/questions/60048249/what-is-the-right-name-of-event-handler-onclick-or-handleclick
꼭 지켜야되는지는 모르겠지만, 한번 고려해보는건 어떨까요?

soo96 replied:
네 고려해 보겠습니다. 감사합니다.🙏🏻

soo96 committed:
refactor: 회원가입 API 요청 바디형식 변경

soo96 committed:
chore: PR 리뷰내용 반영

- Signup 컴포넌트 handleSignup -> handleClick 이벤트명으로 변경
- props 타입정의interface -> type으로 변경

soo96 commented:
@smileboy0014 @leesm0218  리뷰 내용 반영 완료했습니다.🙂

smileboy0014 reviewed (approved):
고생하셨습니다! 👍👍

soo96 committed:
refactor: 닉네임 글자수 검증 에러코드 추가

leesm0218 reviewed (approved):
수고하셨습니다.

soo96 committed:
chore: PR 리뷰 반영 - styles.ts 오탈자 수정
